lib/browser/compiler/core.js

```diff
@@ -103,9 +103,9 @@ riot.compile = function(arg, fn) {
 // reassign mount methods
 var mount = riot.mount
 
-riot.mount = function(a, b, c) {
+riot.mount = function(a, b, c, d) {
```

Comment:
I really like that last code snippet you mentioned.

I hadn't thought of the various signatures, thank you very much for bringing that to my attention.